kiota_serialization_form/form_serialization_writer.py

```diff
@@ -186,24 +187,26 @@ def write_object_value(
             additional_values_to_merge (tuple[Parsable]): The additional values to merge to the
             main value when serializing an intersection wrapper.
         """
-        if key and (value or additional_values_to_merge):
-            temp_writer = self._create_new_writer()
-
-            if value:
-                self._serialize_value(temp_writer, value)
-
-            if additional_values_to_merge:
-                for additional_value in filter(lambda x: x is not None, additional_values_to_merge):
-                    self._serialize_value(temp_writer, additional_value)
-                    if on_after := self.on_after_object_serialization:
-                        on_after(additional_value)
-
-            if value and self._on_after_object_serialization:
-                self._on_after_object_serialization(value)
-
-            if len(self.writer) > 0:
-                self.writer += "&"
-            self.writer += f"{quote_plus(key.strip())}={temp_writer.writer}"
+        if self.depth > 0:
+            raise Exception("Form serialization does not support nested objects.")
+        self.depth += 1
+        temp_writer = self._create_new_writer()
+
+        if value is not None:
+            self._serialize_value(temp_writer, value)
+
+        for additional_value in filter(lambda x: x is not None, additional_values_to_merge):
+            self._serialize_value(temp_writer, additional_value)
+            if on_after := self.on_after_object_serialization:
+                on_after(additional_value)
+
+        if value and self._on_after_object_serialization:
+            self._on_after_object_serialization(value)
+
+        if len(self.writer) > 0:
```

Comment:
Could we write unit tests for this? The `temp_writer` initialised doesn't seem to update `self.writer`. Perhaps `serialize_value` could directly write to `self`?